### PR TITLE
adds kuro laser guns

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
adds kuro laser guns go look at rens doc pretty much the same other than the unwielded accuracy is worse overall other than pistol because akimbo bad

## Why It's Good For The Game
cool new laser guns are pretty damn cool

## Changelog
:cl:
add: Added new marine laser guns, they shoot blue and have cool stuff
del: tx73 because it sucked
/:cl:

### CODE REVIEW

File: code/modules/projectiles/magazines/energy.dm [hunk 1]
```diff
@@ -91,3 +91,20 @@
 /obj/item/cell/lasgun/lasrifle/update_icon()
 	var/remaining = CEILING((charge / max(maxcharge, 1)) * 100, 25)
 	icon_state = "[base_ammo_icon]_[remaining]"
+
+/obj/item/cell/lasgun/marine
+	name = "\improper TEX Battery"
```
Comment (XSlayer300) [inline code/modules/projectiles/magazines/energy.dm:96]:
```suggestion
	name = "\improper TEX battery"
```

File: code/modules/projectiles/ammo_datums.dm [hunk 2]
```diff
@@ -1349,6 +1349,91 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	hud_state = "laser_disabler"
+	damage = 100
+	penetration = 0
+	damage_type = STAMINA
+
+/datum/ammo/energy/lasgun/pistol/disabler/on_hit_mob(mob/M,obj/projectile/P)
+	staggerstun(M, P, stagger = 0.5, slowdown = 0.75)
+
+/datum/ammo/energy/lasgun/marine/sniper
+	name = "sniper bolt"
```
Comment (XSlayer300) [inline code/modules/projectiles/ammo_datums.dm:1406]:
```suggestion
	name = "focused laser bolt"
```

File: code/modules/projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -1349,6 +1349,91 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	icon_state = "blue_sniper"
+	hud_state = "laser_sniper"
+	damage = 60
+	max_range = 30
+	penetration = 20
+	sundering = 4
+
+/datum/ammo/energy/lasgun/marine/sniper/fire
+	name = "microwave sniper bolt"
```
Comment (XSlayer300) [inline code/modules/projectiles/ammo_datums.dm:1415]:
```suggestion
	name = "microwave focused laser bolt"
```

File: code/modules/projectiles/ammo_datums.dm [hunk 4]
```diff
@@ -1349,6 +1349,91 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	icon_state = "microwavelaser" //Yes, this is red. I don't care. Something something visual clairty
+	damage = 40
+	penetration = 20
+	sundering = 1
+	flags_ammo_behavior = AMMO_ENERGY|AMMO_INCENDIARY|AMMO_SUNDERING
+
+/datum/ammo/energy/lasgun/marine/mg
+	name = "machinegun bolt"
```
Comment (XSlayer300) [inline code/modules/projectiles/ammo_datums.dm:1423]:
```suggestion
	name = "low-powered laser bolt"
```
What color is that?

File: code/modules/projectiles/ammo_datums.dm [hunk 5]
```diff
@@ -1349,6 +1349,91 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+	icon_state = "blue_mg"
+	damage = 15
+	penetration = 15
+	sundering = 1
+
+/datum/ammo/energy/lasgun/marine/mg/efficent
+	name = "low power machinegun bolt"
```
Comment (XSlayer300) [inline code/modules/projectiles/ammo_datums.dm:1430]:
```suggestion
	name = "harmonized low-powered laser bolt"
```

File: code/modules/projectiles/guns/energy.dm [hunk 6]
```diff
@@ -594,3 +594,393 @@
 	A.update_hud(user)
 
 	return TRUE
+
+// Neat laser gun things
+
+//Ok the rifle
+
+/obj/item/weapon/gun/energy/lasgun/standard_marine
+	name = "\improper TEX-R Lasrifle"
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/energy.dm:603]:
```suggestion
	name = "\improper TEX-R lasrifle"
```

Is "TEX" the same as "Terragov EXperimental" or is it "**T**erragov **E**nergy e**X**perimental?"

File: code/modules/projectiles/guns/energy.dm [hunk 7]
```diff
@@ -594,3 +594,393 @@
...
+	desc = "A TerraGov standard issue Lasrifle with an integrated charge selector for normal and high settings. Uses standardized power cells."
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/energy.dm:604]:
```suggestion
	desc = "A TerraGov standard issue laser rifle with an integrated charge selector for normal and high settings, fires bolts of laser energy. Uses standardized power cells."
```

File: code/modules/projectiles/guns/energy.dm [hunk 8]
```diff
@@ -594,3 +594,393 @@
...
+		var/mob/living/carbon/human/human_user = user
+		if(src == human_user.l_hand)
+			human_user.update_inv_l_hand()
+		else if (src == human_user.r_hand)
+			human_user.update_inv_r_hand()
+
+// other rifle
+
+/obj/item/weapon/gun/energy/lasgun/standard_marine/carbine
+	name = "\improper TEX-C Lascarbine"
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/energy.dm:701]:
```suggestion
	name = "\improper TEX-C lascarbine"
```

File: code/modules/projectiles/guns/energy.dm [hunk 9]
```diff
@@ -594,3 +594,393 @@
...
+	desc = "A TerraGov standard issue Lascarbine with an integrated charge selector for normal and high settings. Uses standardized power cells."
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/energy.dm:702]:
```suggestion
	desc = "A TerraGov standard issue laser firearm with an integrated charge selector for normal and high settings, this variant is a carbine, as such it is easier to handle in a pinch. Uses standardized power cells."
```

File: code/modules/projectiles/guns/energy.dm [hunk 10]
```diff
@@ -594,3 +594,393 @@
...
+		fire_delay -= 1 SECONDS
+		fire_sound = 'sound/weapons/guns/fire/laser.ogg'
+		to_chat(user, "[icon2html(src, user)] You [overcharge? "<B>disable</b>" : "<B>enable</b>" ] [src]'s high power mode.")
+		overcharge = FALSE
+
+// pistole
+
+
+/obj/item/weapon/gun/energy/lasgun/standard_marine/pistol
+	name = "\improper TEX-R Laspistol"
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/energy.dm:780]:
```suggestion
	name = "\improper TEX-R laspistol"
```

File: code/modules/projectiles/guns/energy.dm [hunk 11]
```diff
@@ -594,3 +594,393 @@
...
+	desc = "A TerraGov standard issue Laspistol with an integrated charge selector for normal and disabler settings. Uses standardized power cells."
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/energy.dm:781]:
```suggestion
	desc = "A TerraGov standard issue laser pistol with an integrated charge selector for normal and disabler settings. This is a compact model, allowing extreme portability in a rather lethal package, though it remains to be a sidearm for some who prefer the 'pew pew'. Uses standardized power cells."
```

File: code/modules/projectiles/guns/energy.dm [hunk 12]
```diff
@@ -594,3 +594,393 @@
...
+		ammo = GLOB.ammo_list[/datum/ammo/energy/lasgun/marine/pistol]
+		fire_delay -= 0.15 SECONDS
+		fire_sound = 'sound/weapons/guns/fire/laser.ogg'
+		to_chat(user, "[icon2html(src, user)] You [overcharge? "<B>disable</b>" : "<B>enable</b>" ] [src]'s disabler mode.")
+		overcharge = FALSE
+
+//Ok the rifle
+
+/obj/item/weapon/gun/energy/lasgun/standard_marine/sniper
+	name = "\improper TEX-S Sniper"
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/energy.dm:847]:
```suggestion
	name = "\improper TEX-S laser strike rifle"
```

File: code/modules/projectiles/guns/energy.dm [hunk 13]
```diff
@@ -594,3 +594,393 @@
...
+	desc = "A TerraGov standard issue Lassniper with an integrated charge selector for normal and high settings. Uses standardized power cells."
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/energy.dm:848]:
```suggestion
	desc = "A specialized energy sniper rifle for TerraGov marines who prefer long-ranged shooting with dangerous accuracy. Primarily known as the 'strike rifle' for its energy capabilities in the marksmanship role. This comes with an integrated charge selector for normal and high settings. Uses standardized power cells."
```

File: code/modules/projectiles/guns/energy.dm [hunk 14]
```diff
@@ -594,3 +594,393 @@
...
+		ammo = GLOB.ammo_list[/datum/ammo/energy/lasgun/marine/sniper]
+		fire_delay -= 0.5 SECONDS
+		fire_sound = 'sound/weapons/guns/fire/laser.ogg'
+		to_chat(user, "[icon2html(src, user)] You [overcharge? "<B>disable</b>" : "<B>enable</b>" ] [src]'s high power mode.")
+		overcharge = FALSE
+
+//Ok the mg
+
+/obj/item/weapon/gun/energy/lasgun/standard_marine/mg
+	name = "\improper TEX-MG Machinegun"
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/energy.dm:921]:
```suggestion
	name = "\improper TEX-MG autolaser"
```
Autolaser or repeater, take your pick.

File: code/modules/projectiles/guns/energy.dm [hunk 15]
```diff
@@ -594,3 +594,393 @@
...
+	desc = "A TerraGov standard issue Laser MG with an integrated charge selector for normal and low power settings. Uses standardized power cells."
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/energy.dm:922]:
```suggestion
	desc = "A lightweight TerraGov energy machinegun for those who wanted a light show at the best way possible. This comes with a charge selector for normal or harmonized settings, the latter being a low-powered mode which is ideal for long-term suppression. Uses standardized power cells."
```

**Comment by XSlayer300** [comment]
If the lasers are blue, I might as well rename them to 'particle' lasers instead, since they're not considered as 'lasers' anymore.

File: code/modules/projectiles/guns/energy.dm (hunk 14 above)
Comment (Pariah919) [reply code/modules/projectiles/guns/energy.dm:921]: no

File: code/modules/projectiles/guns/energy.dm (hunk 12 above)
Comment (Pariah919) [reply code/modules/projectiles/guns/energy.dm:847]: no

**Review by SplinterGP — CHANGES_REQUESTED** [review]
shitcode full of copypasta

**Comment by Pariah919** [comment]
david said he'd do it better

**Pull request closed without merging**